packages/adapter/addon/rest.ts

```diff
@@ -1063,8 +1117,8 @@ class RESTAdapter extends Adapter.extend(BuildURLMixin) {
     @param {Object} options
     @return {Object}
   */
-  ajaxOptions(url, method, options) {
-    options = assign(
+  ajaxOptions(url: string, method: string, options: JQueryAjaxSettings | RequestInit): Dict<any> {
+    let reqOptions: JQueryRequestInit | FetchRequestInit = assign(
```

Comment:
I took another look at how to type cast this to avoid forcing the cast below. On the face of it, since all these options are optional, I'm not sure we have a reliable way to detect.

https://developer.mozilla.org/en-US/docs/Web/API/Request/Request#Parameters
https://api.jquery.com/jquery.ajax/#jQuery-ajax-settings